### DISCUSSION TITLE
Way to login / register customers

### DISCUSSION THREAD

**Discussion by de-raaf-media** [duncanmcclean/simple-commerce#946] [Help]
Hi there,

Does anybody know of a way to hook into the order process after clicking on checkout? Ik would like to have a page where a customer can register (or login). If there already is an order Ik would like to update the orders customer. I thought PreCheckout would be an event I could use but that is too late in the process. Any ideas?

Mark

**Comment by duncanmcclean** [comment]
Have you seen the [`PostCheckout`](https://simple-commerce.duncanmcclean.com/events#content-postcheckout) event?

**Reply by de-raaf-media** [reply]
Yes I did. But this is even later in the process? I would like to add an option on the /checkout/information page (from the starter kit) to login. If I land on this page the order is already created without a customer connected. So if a user decides to login, how do I update the current orders customer?

**Reply by duncanmcclean** [reply]
> Yes I did. But this is even later in the process? I would like to add an option on the /checkout/information page (from the starter kit) to login. 

To login/register users, you'd need to use Statamic's built-in [`{{ user:login_form }}`](https://statamic.dev/tags/user-login_form#content) / [`{{ user:register_form }}`](https://statamic.dev/tags/user-register_form#overview) tags on the page. 

When you're registering users, you can include inputs customer informations inside the `{{ user:register_form }}`.

For logged in users, you'd need to update the customer's information via the `{{ sc:cart:update }}` tag, which you can do like this ([documented here](https://simple-commerce.duncanmcclean.com/tags/cart#content-update-cart)):

```
{{ sc:cart:update }}
    <input type="text" name="customer[name]">
    <input type="email" name="customer[email]">
{{ /sc:cart:update }}
```

Hopefully that helps!

**Comment by de-raaf-media** [comment]
Thanks. Ik got it to work. I already had a login in place. In the form on the first page in checkout I now check if there is a logged in user. In the OrderSaved event I connect this logged in user to the current order. 

If not a guest customer is created.

**Comment by duncanmcclean** [comment]
When Simple Commerce v6 is released (which will happen in the next month or so), the logged in user will be automatically associated as the customer on the order, as long as you're using the ["Users" customer driver](https://simple-commerce.duncanmcclean.com/customers#content-users-driver). 

Related PR: https://github.com/duncanmcclean/simple-commerce/pull/961
